src/main/zabbix/etc/zabbix/scripts/sonicmq_env.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+# Environment settings for running SonicMQ Zabbix scripts
+
+# Location of SonicMQ client libraries
+SONICMQ_JAR_PATH=/opt/zabbix-sonicmq-monitoring/lib
+
+# Installation path of monitor application
+SMQ_MON_INSTALL_PATH=/opt/zabbix-sonicmq-monitoring
+
+# Monitor output file
+SMQ_MON_OUTPUT_FILE=/tmp/sonicmq-monitor-output.json
+
+# Monitor configuration file, configures SonicMQ connection and monitor application
+SMQ_MON_CONFIG_FILE=$SMQ_MON_INSTALL_PATH/config.json
+
+# Maximum monitor output file age in seconds
+SMQ_MON_OUTPUT_MAX_AGE=120
+
+# Run mode: in cron mode scripts do not trigger updates and cron job is expected to be setup for it
+SMQ_MON_CRON_MODE=0
+
+# Monitor application log
+# Define either absolute path or special value "syslog" to redirect to syslog
+SMQ_MON_LOG=syslog
+
+echo $SMQ_MON_LOG
+
+# Function providing common fetch logic around script specific logic parsing the output
+function sonicmq_fetch_stat {
+  if [ $SMQ_MON_CRON_MODE -eq 0 ]; then
```

Comment:
In cron mode, the file lock is not acquired -> there may be simultaneous write and reads from output file, causing partial or wrong results

Reply:
Does it really need to lock for each call? That seems like really bad performance bottleneck.

Reply:
Yes it is. However, is there any other means to make sure that writing is not in half-way when the item value is read from cache? Some atomic move operation perhaps?

Reply:
Well maybe writing output into separate file and renaming it with mv (mv should be atomic enough).

I'll change it to use file lock. I'm a bit worried though how this will perform if we end up monitoring thousands of items (e.g. connections and subscriptions).